PORTAL/tests/test___main__.py

```diff
@@ -41,10 +45,56 @@ def test_show(tmp_path):
 
     with pytest.raises(SystemExit) as exc:
         __main__._parse_and_main(
-            helpers.setup_temp_env(tmp_path) + [
+            helpers.setup_temp_env(tmp_path)
+            + [
                 "show",
             ],
-            __file__
+            __file__,
         )
 
     assert exc.value.code == 1
+
+
+def test_run_bench(tmp_path, monkeypatch):
+    # Just a basic smoke test
+
+    def dummy(*args, **kwargs):
+        return
+
+    monkeypatch.setitem(__main__.COMMANDS, "attach", dummy)
+
+    __main__._parse_and_main(
+        helpers.setup_temp_env(tmp_path)
+        + [
+            "run-bench",
+            "--worker",
+            "mac",
+            "--benchmarks",
+            "deepcopy",
+            "caf63ec5",
+        ],
+        __file__,
+    )
+
+    queue = json.loads(
+        (tmp_path / "BENCH" / "QUEUES" / "mac" / "queue.json").read_text()
+    )
+
+    assert len(queue["jobs"]) == 1
+    assert queue["jobs"][0].endswith("-mac")
+    assert queue["jobs"][0].startswith("req-compile-bench")
+
+    requests_dir = list((tmp_path / "BENCH" / "REQUESTS").iterdir())
+
+    assert len(requests_dir) == 1
+
+    files = sorted(x.name for x in requests_dir[0].iterdir())
```

Comment:
I think this was better as is -- it's asserting that there is only one request created (an earlier iteration of this created a linux one as well because the workerid wasn't being set everywhere it should).

Reply:
Makes sense.  Thanks for clarifying.  It would still be good to check against `reqid` (and simplify a little):

```suggestion
    reqdir, = (tmp_path / "BENCH" / "REQUESTS").iterdir()
    assert reqdir.name == reqid

    files = sorted(x.name for x in reqdir.iterdir())
```

Reply:
Makes sense.